crate/server/src/routes/permissions.rs

```diff
@@ -71,6 +85,24 @@ pub(crate) async fn grant_permission(
     }))
 }
 
+#[post("/permission/list/{user_id}")]
+pub(crate) async fn list_permission(
+    req: HttpRequest,
+    params: web::Path<String>,
+    findex_server: Data<Arc<FindexServer>>,
+) -> ResponseBytes {
+    let user = findex_server.get_user(&req);
+    let user_id = params.into_inner();
+    info!("user {user}: POST /permission/list/{user_id}");
+
+    let permissions = findex_server.db.get_permissions(&user_id).await?;
```

Comment:
Here set the returned permission to `min(user_permission, permission)` to avoid a user to lookup who are the more powerful users.

Reply:
so a user with a `read` permission will retrieve a filtered list of users permissions with on `read` permissions, right?

Reply:
done in c0b06e9fb98bcf2675069c37d75f1ca11e1e5cbe